src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -131,15 +128,32 @@ public SecurityRestFilter(
     public RestHandler wrap(RestHandler original, AdminDNs adminDNs) {
         return (request, channel, client) -> {
             org.apache.logging.log4j.ThreadContext.clearAll();
-            if (!checkAndAuthenticateRequest(request, channel)) {
-                User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-                boolean isSuperAdminUser = userIsSuperAdmin(user, adminDNs);
-                if (isSuperAdminUser
-                    || (whitelistingSettings.checkRequestIsAllowed(request, channel, client)
-                        && allowlistingSettings.checkRequestIsAllowed(request, channel, client))) {
-                    if (isSuperAdminUser || authorizeRequest(original, request, channel, user)) {
-                        original.handleRequest(request, channel, client);
-                    }
+            final SecurityRequestChannel requestChannel = SecurityRequestFactory.from(request, channel);
```

Comment:
@peternied Do you think it makes sense in the long-term to aim for the security plugin to not have access to the channel at all? I'd like to eventually have a clean interface in core that the security plugin can provide an `earlyResponse` something similar that can then be used to send the response before the original handler is called.

Reply:
I think the concept of an 'early' response shouldn't be part of the RestHandler ecosystem at all - can you get more than one, what is the order they are used ? is a good example of a problem to avoid

However, in the context of the HeaderValidator and the classes inside of core that use it, it might make sense to check for an early response.